statsmodels/tsa/statespace/_initialization.pyx.in

```diff
@@ -0,0 +1,248 @@
+#cython: boundscheck=False
+#cython: wraparound=False
+#cython: cdivision=False
+"""
+State Space Models - Initialization
+
+Author: Chad Fulton  
+License: Simplified-BSD
+"""
+
+{{py:
+
+TYPES = {
+    "s": ("np.float32_t", "np.float32", "np.NPY_FLOAT32"),
+    "d": ("np.float64_t", "float", "np.NPY_FLOAT64"),
+    "c": ("np.complex64_t", "np.complex64", "np.NPY_COMPLEX64"),
+    "z": ("np.complex128_t", "complex", "np.NPY_COMPLEX128"),
+}
+
+}}
+
+# Typical imports
+import numpy as np
+import warnings
+cimport numpy as np
+cimport cython
+
+np.import_array()
+
+from statsmodels.src.math cimport *
+cimport scipy.linalg.cython_blas as blas
+cimport scipy.linalg.cython_lapack as lapack
+cimport statsmodels.tsa.statespace._tools as tools
+
+{{for prefix, types in TYPES.items()}}
+from statsmodels.tsa.statespace._representation cimport {{prefix}}Statespace
+{{endfor}}
+
+cdef int FORTRAN = 1
+
+{{for prefix, types in TYPES.items()}}
+{{py:cython_type, dtype, typenum = types}}
+{{py:
+combined_prefix = prefix
+combined_cython_type = cython_type
+if prefix == 'c':
+    combined_prefix = 'z'
+    combined_cython_type = 'np.complex128_t'
+if prefix == 's':
+    combined_prefix = 'd'
+    combined_cython_type = 'np.float64_t'
+}}
+
+## State Space Initialization
+cdef class {{prefix}}Initialization(object):
+
+    def __init__(self, int k_states, {{cython_type}} [:] constant,
+                 {{cython_type}} [::1, :] stationary_cov,
+                 np.float64_t approximate_diffuse_variance=1e6):
+        cdef:
+            int k
+            np.npy_intp dim1[1]
+            np.npy_intp dim2[2]
+
+        self.k_states = k_states
+        self.constant = constant
+        self.stationary_cov = stationary_cov
+        self.approximate_diffuse_variance = approximate_diffuse_variance
+
+        # Validate
+        tools.validate_vector_shape('known constant', &self.constant.shape[0], self.k_states, None)
+        tools.validate_matrix_shape('known covariance', &self.stationary_cov.shape[0], self.k_states, self.k_states, None)
+
+        # Internal temporary matrices
+        dim2[0] = self.k_states; dim2[1] = self.k_states;
+        self._tmp_transition = np.PyArray_ZEROS(2, dim2, {{typenum}}, FORTRAN)
+        dim2[0] = self.k_states; dim2[1] = self.k_states;
```

Comment:
flake8; please split up; easier to read;

Reply:
Good idea.